### PR TITLE
Added exceptions for Authentication and Management\User

### PR DESCRIPTION
**Pull request by Nyholm**
I also created a shared base class between Authentication and Management. 

Related to #190

### CODE REVIEW

**Comment by Nyholm** [comment]
I would like to merge this PR and then a second PR with the rest of the exceptions. After that I think we are ready for another pre-release.

File: src/API/Management/Users.php [hunk 1]
```diff
@@ -84,30 +130,46 @@ public function delete($userId)
...
-        return ResponseMediator::getContent($response);
+        if (200 === $response->getStatusCode()) {
+            return ResponseMediator::getContent($response);
+        }
+
+        $this->handleExceptions($response);
     }
 
     /**
+     * TODO find a link to the documentation
```
Comment (cocojoe) [inline src/API/Management/Users.php]: @glena this still relevant, can't see it under https://auth0.com/docs/api/management/v2#!/Users/

Comment (glena) [inline src/API/Management/Users.php]: the idea was to point to any doc about refresh tokens explaining token revocation. 

This will be deprecated in favor of the new `/oauth/revoke` one I think.

**Comment by Nyholm** [comment]
Thank you

File: src/API/Management/Users.php (hunk 1 above)
Comment (cocojoe) [inline src/API/Management/Users.php]: @glena should we drop this then as it's a deprecated end point? If it's not in management API docs then I feel it shouldn't be in the SDK.

Comment (Nyholm) [inline src/API/Management/Users.php]: I agree that it should be dropped if it is not in the management API. But that should be a separate PR.

Comment (glena) [inline src/API/Management/Users.php]: yes

**Comment by Nyholm** [comment]
Thank you for merging.
